ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PostgresEventStrategy.scala

```diff
@@ -7,45 +7,72 @@ import com.daml.platform.store.appendonlydao.events.Party
 import com.daml.platform.store.backend.EventStorageBackend.FilterParams
 import com.daml.platform.store.backend.common.ComposableQuery.{CompositeSql, SqlStringInterpolation}
 import com.daml.platform.store.backend.common.EventStrategy
+import com.daml.platform.store.interning.StringInterning
 
 object PostgresEventStrategy extends EventStrategy {
   override def filteredEventWitnessesClause(
       witnessesColumnName: String,
       parties: Set[Party],
-  ): CompositeSql =
-    if (parties.size == 1)
-      cSQL"array[${parties.head.toString}]::text[]"
-    else {
-      val partiesArray: Array[String] = parties.view.map(_.toString).toArray
-      cSQL"array(select unnest(#$witnessesColumnName) intersect select unnest($partiesArray::text[]))"
-    }
+      stringInterning: StringInterning,
+  ): CompositeSql = {
+    val internedParties: Array[java.lang.Integer] = parties.view
+      .flatMap(party => stringInterning.party.tryInternalize(party).map(Int.box).toList)
+      .toArray
+    if (internedParties.length == 1)
+      cSQL"array[${internedParties.head}]::integer[]"
+    else
+      cSQL"array(select unnest(#$witnessesColumnName) intersect select unnest($internedParties::integer[]))"
+  }
 
   override def submittersArePartiesClause(
       submittersColumnName: String,
       parties: Set[Party],
+      stringInterning: StringInterning,
   ): CompositeSql = {
-    val partiesArray = parties.view.map(_.toString).toArray
-    cSQL"(#$submittersColumnName::text[] && $partiesArray::text[])"
+    val partiesArray: Array[java.lang.Integer] = parties.view
+      .flatMap(party => stringInterning.party.tryInternalize(party).map(Int.box).toList)
+      .toArray
+    cSQL"(#$submittersColumnName::integer[] && $partiesArray::integer[])"
   }
 
   override def witnessesWhereClause(
       witnessesColumnName: String,
       filterParams: FilterParams,
+      stringInterning: StringInterning,
   ): CompositeSql = {
     val wildCardClause = filterParams.wildCardParties match {
       case wildCardParties if wildCardParties.isEmpty =>
         Nil
 
       case wildCardParties =>
-        val partiesArray = wildCardParties.view.map(_.toString).toArray
-        cSQL"(#$witnessesColumnName::text[] && $partiesArray::text[])" :: Nil
+        val partiesArray: Array[java.lang.Integer] = wildCardParties.view
+          .flatMap(party => stringInterning.party.tryInternalize(party).map(Int.box).toList)
+          .toArray
+        if (partiesArray.isEmpty)
```

Comment:
Why can `partiesArray` come out as `isEmpty` for `witnessesWhereClause` but not for `filteredEventWitnessesClause`

Reply:
sharp eye!
It could based on the type.
That would boil down passing empty array down.
This can only happen, if we have never seen any of those parties which was the argument:
* This type of queries are quite futile because they will never get back any result: the visibility filter is controlled by the same set of parties (= if that is empty -> no result)
* But they still need to be valid SQLs
* Front-running next question from this: for the OracleEventStrategy there is a empty handling there because the parameter is not passed as an array but a anorm-multivalue IN clause, which if empty produces invalid SQL
* Front-running possible suggestion for improvement: it is possible in the aforementioned circumstance to not do the query at all! But it would require a slight re-wamping of the functions and interfaces here. Also the improvement is not obviously huge, clients probably not banging the LAPI service with futile queries, or not that much that it is worth optimizing for (because of that reason I was not even creating an improvement task for this neither)